### PR TITLE
First pass at my proposed retry/housekeeping stuff! No notification stuff... YET

### PR DESCRIPTION
**Pull request by pobocks**
Primary changes include:
- move requirements into requirements.txt (requirements cover ALL scripts in dir)
- optionally pass in config file via script arguments
- optionally specify output_basedir in yaml config file
- update usage of yaml to be compatible with ruamel.yaml version > 18
- retry JSON fetching via tenacity - make values configurable in next pass?
- updates to validation checks to simplify and provide more coverage
- move main function of script into main function run if __name__ == "__main__"
- removed trailing whitespace because my editor is configured to do that automatically

### CODE REVIEW

**Review by Wiscmapper — APPROVED** [review]

**Review by Wiscmapper — APPROVED** [review]

File: esriscanner.py [hunk 1]
```diff
@@ -376,6 +378,7 @@ def main():
 
     # default log level is info
     levelname = theDict.get('log_level', 'INFO').upper()
+    print(levelname)
```
Comment (pobocks) [inline esriscanner.py:381]: Maybe make this `log.info(levelname)`

File: esriscanner.py [hunk 2]
```diff
@@ -351,7 +353,7 @@ def add_to_report(message):
     print(message, file=report_target)
 
 log = None
-produce_report = False
+produce_report = True
```
Comment (pobocks) [inline esriscanner.py:356]: The reason this is set to False initially is that (as I understood it) you only want the report or email in cases where you encounter an error - the idea is that any error reported gets paired with setting this var.

**Review by pobocks — COMMENTED** [review]
Minor comments

File: esriscanner.py (hunk 2 above)
Comment (Wiscmapper) [reply esriscanner.py:356]: Oh, oops, I misunderstood.  I thought this had to be set to True in order to trigger reporting.

File: esriscanner.py (hunk 1 above)
Comment (Wiscmapper) [reply esriscanner.py:381]: Got it, will change.

**Pull request merged**